### ISSUE TITLE
[TF 2.0] tf.keras.losses duplication?

### ISSUE THREAD

**Issue by danaugrs** [tensorflow/tensorflow#26012]
**Windows 10
TF Version: b'v1.11.0-rc2-4-gc19e29306c' 1.11.0
Anaconda Python 3.6.5
GPU: GeForce GTX 1070 Max-Q Design
Tensorflow 2.0 (gpu) preview installed via pip.**

I'm building a [reinforcement learning framework](https://github.com/danaugrs/huskarl) on top of TensorFlow 2.0 using the `tf.keras` API and I've come across the following issue.

The 2.0 API docs for [`tf.keras.losses`](https://www.tensorflow.org/versions/r2.0/api_docs/python/tf/keras/losses) shows many objects that are not actually available in the preview package. For example the loss classes such as Huber. Hinge, etc... are not accessible.

1. Why are those classes not included in the preview package?
2. Why are there both classes and functions for many of the same loss types? That seems like unnecessary duplication.
2a. Why is there a `Huber` class but no `huber` function?
3. I'd love to contribute PRs and help fix these issues. Would that be desired?

Edit: This has also been noticed here: https://github.com/tensorflow/tensorflow/issues/26007

**Comment by dynamicwebpaige** [comment]
@danaugrs - Adding @martinwicke and @fchollet -- and we'd love to have help resolving the issues you detailed in the API documentation. 

PS: thanks for the great work on Huskarl! :slightly_smiling_face:

**Comment by martinwicke** [comment]
It looks from the description like this is regarding 1.11 (`TF Version: b'v1.11.0-rc2-4-gc19e29306c' 1.11.0`)? 

This is expected, this was fixed in later versions (it wasn't ready back in late Summer 2018 when 1.11 was cut). I'll close this.

@pavithrasv FYI

**Comment by danaugrs** [comment]
@martinwicke Thanks, I must have gotten my virtual envs confused - apologies. I updated in any case and the docs match what I see. However, with the latest version (`tf.__version__ == '2.0.0-dev20190214'`) executing `pip show tensorflow` still shows `Version: 1.11.0`. This is preventing my library from requiring TF 2.0 explicitly. Is there a reason the PyPI version hasn't been updated?

As a heads up, `tf.GIT_VERSION`, `tf.VERSION` don't exist in the latest version, so the "Bug/Performance Issue" template instructions don't work in this case.

Also, would someone be able to answer questions 2 and 2a ?

> 2. Why are there both classes and functions for many of the same loss types? That seems like unnecessary duplication.
>     2a. Why is there a `Huber` class but no `huber` function?

**Comment by martinwicke** [comment]
That's because the preview is a different package (tf-nightly-2.0-preview).
So technically you haven't upgraded the TensorFlow package as far as pip is
concerned.

**Comment by pavithrasv** [comment]
=> 2. The functions compute per sample loss value. The classes call the functions in them internally and also apply weights and reduce the loss value to a scalar. 
=> There is a `huber` function. It is called `huber_loss`. May be the the `_loss` suffix in the name can be removed.

**Comment by danaugrs** [comment]
@pavithrasv [huber_loss](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/losses.py#L813) exists but as you can see it's not exported and therefore not accessible. `logloss` is also not exported but IMHO should be. I made a PR to address these issues: https://github.com/tensorflow/tensorflow/pull/26034

I also agree that `huber_loss` should be renamed to `huber` for consistency.

**Comment by pavithrasv** [comment]
Oh yes, we can definitely export those, thank you for the PR!
